sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-supplier-integration.scheduler.e2e-spec.ts

```diff
@@ -366,6 +368,220 @@ describe(describeProcessorRootTest(QueueNames.CASSupplierIntegration), () => {
     },
   );
 
+  it("Should create a new site and update the student CAS supplier when an active CAS supplier exists with no match addresses.", async () => {
+    // Arrange
+    const referenceDate = new Date();
+    const savedCASSupplier = await saveFakeCASSupplier(db);
+    // Configure CAS mock to return a result for the GetSupplier
+    // with the same supplier number and address line 1 from the
+    // saved CAS supplier but a different postal code.
+    casServiceMock.getSupplierInfoFromCAS = jest.fn(() =>
+      Promise.resolve(
+        createFakeCASSupplierResponse({
+          initialValues: {
+            supplierNumber: savedCASSupplier.supplierNumber,
+            addressLine1: savedCASSupplier.supplierAddress.addressLine1,
```

Comment:
The initial values will be useful here unlike there other cases, the reason being we apply formatting on the address  line city and postal code before we send it to CAS and if the country is not CA(we are lucky that the create fake student is setting country to Canada) it will not even make the get supplier call.

I would recommend saving the student in the way similar to the existing test case. 

![image](https://github.com/user-attachments/assets/22bb9b75-2537-454d-8759-dd62c9d5c278)